packages/flutter_app/lib/router/router_provider.dart

```diff
@@ -13,26 +14,21 @@ Raw<GoRouter> router(RouterRef ref) {
   final initialization = ref.watch(initializationProvider);
   return GoRouter(
     routes: [
-      GoRoute(
-        path: '/init',
-        pageBuilder: (context, state) {
-          return const NoTransitionPage(child: InitializationPage());
-        },
-      ),
       ...$appRoutes,
     ],
     redirect: (context, state) {
       if (initialization.isLoading || initialization.hasError) {
-        return '/init';
+        return InitializationRoute.path;
       }
       if (state.matchedLocation == '/') {
-        return HomeShellBranchData.path;
+        return HomeRouteData.path;
       }
       return null;
     },
     errorBuilder: (context, state) => ErrorRoute(
       location: state.uri.path,
       exception: state.error!,
     ).build(context, state),
+    observers: ref.watch(trackerProvider).navigatorObservers(),
```

Comment:
Send screen tracking using FirebaseAnalyticsObserver.

Reply:
![badge](https://img.shields.io/badge/review-imo-important) 
I thought that `observers` would not fire in the current `StatefulShellRoute` 👀

Reference: https://github.com/flutter/flutter/issues/112196

I tried setting it as below to monitor.
The nameExtractor callback should always be executed when didPush or didPop of FirebaseAnalyticsObserver is executed, but the callback of nameExtractor was not entered during tab transitions or transitions within tabs.

```dart
    observers: ref.watch(trackerProvider).navigatorObservers(
      nameExtractor: (settings) {
        final name = settings.name;
        return name;
      },
    ),
```

[In another project](https://github.com/altive/medeta/blob/main/medeta_app/lib/router/router_provider.dart#L54), I am using `router.routerDelegate.addListener` as a workaround.

Reply:
@naipaka 
Thanks for the link to the very helpful project!
I implemented the same way!

For some reason it still doesn't show up in DebugView, but I was able to see the log output in the console!
Please review!
[04e7b57](https://github.com/altive/flutter_app_template/pull/452/commits/04e7b5779e4bc94b6afbe4ad3befd0818fc82e3f)

![image](https://github.com/altive/flutter_app_template/assets/8737743/b6d07ee1-afa4-4a2e-a753-abd770e59ac8)

Reply:
I want to send the `settings.name` as it is, so we use `defaultNameExtractor` !
The content of the path to be sent has been changed to `/{last of pathSegments}`!
[b3a61fc](https://github.com/altive/flutter_app_template/pull/452/commits/b3a61fc6acd9e12571f35e1aa2fc4eb3124832ed)